articles/user-guide/testing-debugging.md

```diff
@@ -34,40 +34,73 @@ In the upper right corner, search for `Q#`, and select the **Q# Test Project** t
+Any Q# operation or function that takes an argument of type `Unit` and returns `Unit` can be marked as a unit test via the `@Test("...")` attribute.
+In the previous example, the argument to that attribute, `"QuantumSimulator"`, specifies the target on which the test runs. A single test can run on multiple targets. For example, add an attribute `@Test("ResourcesEstimator")` before `AllocateQubit`.
+
 ```qsharp
     @Test("QuantumSimulator")
     @Test("ResourcesEstimator")
     operation AllocateQubit () : Unit {
         ...
 ```
-Save the file and run all tests. There should now be two unit tests, one where `AllocateQubit` runs on the `QuantumSimulator`, and one where it runs in the `ResourcesEstimator`. 
 
-The Q# compiler recognizes the built-in targets `"QuantumSimulator"`, `"ToffoliSimulator"`, and `"ResourcesEstimator"` as valid run targets for unit tests. It is also possible to specify any fully qualified name to define a custom run target. 
+After saving the file you will see two unit tests when running the tests: one where `AllocateQubit` runs on the `QuantumSimulator`, and one where it runs in the `ResourcesEstimator`.
+
+The Q# compiler recognizes the built-in targets `"QuantumSimulator"`, `"ToffoliSimulator"`, and `"ResourcesEstimator"` as valid run targets for unit tests. It is also possible to specify any fully qualified name to define a custom run target.
+
+Besides the code file, the test project template will populate the `.csproj` file with the following parameters:
+
+```xml
+<Project Sdk="Microsoft.Quantum.Sdk/0.16.2105140472">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Quantum.Xunit" Version="0.16.2105140472" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.4.0" />
+    <PackageReference Include="xunit" Version="2.4.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1" />
+    <DotNetCliToolReference Include="dotnet-xunit" Version="2.3.1" />
+  </ItemGroup>
+
+</Project>
+```
+
+The first line specifies the version number of the software development kit used to build the application. Note that the exact version numbers you see will depend on your installation.
+
+The `TargetFramework` generally contains either of two values for Q# applications depending on the project type: `netcoreapp3.1` for executable projects, and `netstandard2.1` for libraries. Next, the `IsPackable` parameter is set to false (true when omitted). It determines whether a NuGet package is generated from this project when the [`dotnet pack`](/dotnet/core/tools/dotnet-pack) command is run.
+
+Lastly, the file contains all the NuGet package dependencies required to run unit tests. Make sure to include these if you are creating a test project manually or are converting a regular project.
```

Comment:
It would be helpful to be more specific here about which NuGet packages serve what purpose. For example, if a person is creating a test project for command line or for VS Code, do they need NuGet package "xunit.runner.visualstudio"?

Reply:
Good me point, let me look at that.

Reply:
From my testing it appears that both `xunit.runner.visualstudio` and `dotnet-xunit` are required to run tests from the command line, but only `xunit.runner.visualstudio` is necessary to run tests from within Visual Studio.
I'm adding the info I gathered so far, but if I obtain more I will update with a new PR.